### DISCUSSION TITLE
How to update v86 save state to new media

### DISCUSSION THREAD

**Discussion by Darin755** [copy/v86#685] [Q&A]
Hello,
I am trying to update a saved state with a new media without losing the files stored in the 9p filesystem. My first thought was to just reboot the emulator and then save a new save state. That does not work because v86 remembers the old boot media. Is there a way to update to a newer version of the media without data loss?

**Comment by hello-smile6** [comment]
Darin755: Get your VM on a network and send your files using FTP.

**Reply by Darin755** [reply]
That would work for a single use case. However I am working on a project (browser linux) and I was looking for a way to update the iso while preserving the 9p fs for anyone who uses the project

**Reply by hello-smile6** [reply]
Darin755: You might be able to get data from the v86state file. The filesystem should be in the first line output by `strings`, somewhere in the JSON.

**Reply by Darin755** [reply]
I think the best answer is to get all of the files and store them temporarily while it reloads and then create all of them again.

**Reply by Darin755** [reply]
So I can use emulator.fs9p.get_state() to get the sate but emulator.fs9p.set_state(state) throws the folowing error: Uncaught TypeError: a[2] is not iterable
I will continue to look in to it.
